megatron/timers.py

```diff
@@ -303,7 +303,7 @@ def write(self, names, writer, iteration, normalizer=1.0,
         assert normalizer > 0.0
         name_to_min_max_time = self._get_global_min_max_time(
             names, reset, barrier, normalizer)
-        if writer is not None:
+        if writer.is_enabled():
             for name in name_to_min_max_time:
                 _, max_time = name_to_min_max_time[name]
-                writer.add_scalar(name + '-time', max_time, iteration)
+                writer.add_scalar_to_tb(name + '-time', max_time, iteration)
```

Comment:
will this work for both tensorboard and wandb ?

Reply:
Yes Work's fine and Well Optimized for everything

Reply:
Hi Jyothendra, I fixed this comments by invoking add_scalar directly as it fits in both tensorboard and wandb.
<img width="684" alt="8bc97d0cc9d4b2218e5f1f4ebac2f1f" src="https://github.com/microsoft/Megatron-DeepSpeed/assets/96406262/fd8aae43-a3fb-435f-8c1f-6b0919a4ac57">